sensor/tests/resource/helper.go

```diff
@@ -324,6 +325,37 @@ func (c *TestContext) LastViolationStateWithTimeout(name string, assertion Alert
 
 }
 
+// LastViolationStateByID checks the violation state by deployment ID
+func (c *TestContext) LastViolationStateByID(id string, assertion AlertAssertFunc, message string, checkEmptyAlertResults bool) {
+	c.LastViolationStateByIDWithTimeout(id, assertion, message, checkEmptyAlertResults, 3*time.Second)
+}
+
+// LastViolationStateByIDWithTimeout checks that a violation state for a deployment must match `assertion`. If violation state does not match
+// until `timeout` the test fails.
+func (c *TestContext) LastViolationStateByIDWithTimeout(id string, assertion AlertAssertFunc, message string, checkEmptyAlertResults bool, timeout time.Duration) {
+	timer := time.NewTimer(timeout)
+	ticker := time.NewTicker(10 * time.Millisecond)
+	var lastErr error
+	for {
+		select {
+		case <-timer.C:
+			c.t.Fatalf("timeout reached waiting for violation state (%s): %s", message, lastErr)
+		case <-ticker.C:
+			messages := c.GetFakeCentral().GetAllMessages()
+			alerts := GetAllAlertsForDeploymentID(messages, id, checkEmptyAlertResults)
+			var lastViolationState *central.AlertResults
+			if len(alerts) > 0 {
+				lastViolationState = alerts[len(alerts)-1].GetEvent().GetAlertResults()
```

Comment:
This can panic on `nil` if you run this before any alerts for this deployment ID are sent. 
I suggest to check on `nil` and `continue` if there are no alerts to assert against.

Reply:
I've stopped using `GetAllAlertsForDeploymentID` so now the code is different here.